Adding Payment Gateway app and CyberSource implementation

jkachel opened the pull request:
Adds a generic Payment Gateway app to facilitate processing payments in other Django apps, and a specific Gateway that supports CyberSource as the payment processor. 

This will be initially used for mitodl/mitxonline#341.

rhysyngsun commented on src/mitol/payment_gateway/api.py:
```diff
@@ -0,0 +1,206 @@
+        """
+        This is the entrypoint to the payment gateway.
+        Args:
+            order: dict containing:
+                - username      Purchaser username
+                - ip_address    Purchaser's IP address
+                - reference     Order reference number
+            cart: list of dicts containing:
+                - code      Item code         (in xPro: content_type)
+                - name      Item name         (in xPro: description)
+                - quantity  Quantity          (in xPro: quantity)
+                - sku       SKU               (in xPro: content_object.id)
+                - taxable   Taxable amount    (in xPro: 0)
+                - unitprice Unit price        (in xPro: calculated price with discount)
```
I'm thinking of a few improvements here:

- Remove `cart` and just nest a list of `items` under `order`
- Instead of `dict` objects, use dataclasses:
  - As implemented a `dict` would actually cause an error because the code in `CyberSourcePaymentGateway` accesses values by property instead of by key.
  - Dataclasses would provide the best of both worlds - both flexible data structure like a dict (via subclassing) and providing a typed interface. We probably won't use the extensibility of this quite yet, but it'll be good to have the right data type.

jkachel commented:
Fixed all the cleanup things and refactored to use dataclasses in 6f7f5eedddc27758a852d4c8f639cd9a8dd0d1f6. Also bumped the pants version a bit (to 2.7.0.dev4).

jkachel commented:
FYI - latest release incorporates the global Pants update.
